src/main/scala/com/datastax/gatling/plugin/utils/ResultSetUtils.scala

```diff
@@ -0,0 +1,77 @@
+package com.datastax.gatling.plugin.utils
+
+import java.util.concurrent.CompletionStage
+
+import com.datastax.dse.driver.api.core.graph.{AsyncGraphResultSet, GraphNode}
+import com.datastax.oss.driver.api.core.cql.{AsyncResultSet, Row}
+import org.apache.tinkerpop.gremlin.process.traversal.Path
+import org.apache.tinkerpop.gremlin.structure.{Edge, Property, Vertex, VertexProperty}
+
+import collection.JavaConverters._
+
+object ResultSetUtils {
+
+  def asyncResultSetToIterator(resultSet:AsyncResultSet): Iterator[Row] =
+    new AsyncResultSetIterator(
+      new CqlAsyncRS(resultSet))
+
+  def asyncGraphResultSetToIterator(resultSet:AsyncGraphResultSet): Iterator[GraphNode] =
+    new AsyncResultSetIterator(
+      new GraphAsyncRS(resultSet))
+}
+
+object GraphResultSetUtils {
+
+  private def buildFilterAndMapFn[T]
+  (filterFn: GraphNode => Boolean, mapFn: GraphNode => T)
+  (resultSet: AsyncGraphResultSet, key:String):Iterator[T] = {
+    ResultSetUtils.asyncGraphResultSetToIterator(resultSet )
+      .map(graphNode => graphNode.getByKey(key))
+      .filter(filterFn)
+      .map(mapFn)
+  }
+
+  def edges: (AsyncGraphResultSet, String) => Iterator[Edge] = buildFilterAndMapFn(_.isEdge, _.asEdge)
```

Comment:
Maybe instead of propagating this tuple to the client, we can create a dedicated case class?:
```
case class AsyncResultSetWithKey(resultSet: AsyncGraphResultSet, key:String)
```
with such a solution there will be no ambiguity regardless the String parameter

Reply:
I'm not sure I follow.  AsyncGraphResultSet and the key string here are args to the edges function... which was intended to preserve the existing API.

Reply:
I mean collecting those two arguments into one case class, but you are right - probably not worth the effort